distributed/scheduler.py

```diff
@@ -5220,6 +5230,27 @@ def remove_plugin(self, plugin):
         """Remove external plugin from scheduler"""
         self.plugins.remove(plugin)
 
+    def register_scheduler_plugin(self, comm=None, plugin=None):
```

Comment:
Let's make this handler `async` (similar to `register_worker_plugin`) which will allow us to directly `await` `plugin.start` below instead of using `self.sync`:

```suggestion
    async def register_scheduler_plugin(self, comm=None, plugin=None):
```